### DISCUSSION TITLE
Public key "pkcs8" must be PKCS#8 formatted string

### DISCUSSION THREAD

**Discussion by barrylavides** [panva/jose#616] [Q&A] [answered]
Hi, I can already generate a signature w/ rsa key pair in pkcs8 format. Here's the [reference](https://kb.vander.host/security/how-to-generate-rsa-public-and-private-key-pair-in-pkcs8-format/)

Here's my code to generate the signature
```
const fs = require("fs");
const jose = require("jose");

const test = async function () {
    const alg = "RS256";
    const pkcs8 = fs.readFileSync("./test/pkcs8.key", "utf8");
    const privateKey = await jose.importPKCS8(pkcs8, alg);

    const jws = await new jose.FlattenedSign(
        new TextEncoder().encode("detached payload example")
    )
        .setProtectedHeader({ alg })
        .sign(privateKey);

    console.log("jws: ", jws);
};
(async function () {
    await test();
})();
```

Problem is I can't verify it, error is saying my public key is not in pkcs8 format

`TypeError: "pkcs8" must be PKCS#8 formatted string`

```
const pbcs8 = fs.readFileSync("./test/publickey.crt", "utf8");
const publicKey = await jose.importPKCS8(pbcs8, alg);

const { payload, protectedHeader } = await jose.flattenedVerify(
    jws,
    publicKey
);
```

Is my key pair generation incorrect? How can I import a public key to verify a signature?

**Comment by panva** [comment, marked as answer]
> How can I import a public key to verify a signature?

- [Public Key Import (SPKI)](https://github.com/panva/jose/blob/main/docs/functions/key_import.importSPKI.md)
- [Public Key Import (X.509 Certificate)](https://github.com/panva/jose/blob/main/docs/functions/key_import.importX509.md)
